pennylane/devices/tests/test_measurements.py

```diff
@@ -208,6 +209,40 @@ def circuit():
 
         assert np.allclose(res, expected, atol=tol(dev.shots))
 
+    def test_projector_expectation(self, device, tol):
+        """Test that arbitrary Projector expectation values are correct"""
+        n_wires = 2
+        dev = device(n_wires)
+
+        if "Projector" not in dev.observables:
+            pytest.skip("Skipped because device does not support the Projector observable.")
+
+        theta = 0.732
+        phi = 0.523
+
+        @qml.qnode(dev)
+        def circuit(basis_state):
+            qml.RY(theta, wires=[0])
+            qml.RY(phi, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.Projector(basis_state, wires=[0, 1]))
+
+        res = circuit([0, 0])
+        expected = (np.cos(phi / 2) * np.cos(theta / 2)) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.shots))
+
+        res = circuit([0, 1])
+        expected = (np.sin(phi / 2) * np.cos(theta / 2)) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.shots))
+
+        res = circuit([1, 0])
+        expected = (np.sin(phi / 2) * np.sin(theta / 2)) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.shots))
+
+        res = circuit([1, 1])
+        expected = (np.cos(phi / 2) * np.sin(theta / 2)) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.shots))
```

Comment:
@wongwsvincent this is a great test 🙌 Did you compute the expected projector expectation by hand?

Reply:
Yes, I did all the calculations by hand. I thought of using Mathematica to do the matrix multiplication for me, but I ended up deciding that it shouldn't take too long to calculate by hand. 😅